### PR TITLE
js-annot: add Provides and Requires annotations to js file

### LINKED ISSUE #132: js_of_ocaml primitives
A couple of issues with the runtime.js file"

1. Functions are not annotated with `// Provides:` and `// Requires:` comments, so the linker does not pick them up

2. I cannot find out where the `string_unsafe_*` and `caml_ba_unsafe_*` functions they call are defined.

### PR DESCRIPTION
**Pull request by seliopou**
Closes #132

### CODE REVIEW

**Comment by seliopou** [comment]
@martindemello can you confirm this fixes your problem?

**Comment by seliopou** [comment]
I see. That's easy to fix.

**Comment by seliopou** [comment]
I pushed another commit. It looks like unsafe versions of bigstring intrinsics aren't available in JavaScript. I just changed the implementation to use the ones that are available.

Could you give it another spin?

File: lib/runtime.js [hunk 1]
```diff
@@ -31,20 +31,26 @@
...
+    caml_ba_set_1(dst, dst_off + i, caml_ba_get_1(src, src_off + i));
   }
 }
 
+//Provides: angstrom_bigstring_blit_from_bytes
+//Requires: caml_ba_set_1, string_unsafe_get
 function angstrom_bigstring_blit_from_bytes(src, src_off, dst, dst_off, len) {
   for (var i = 0; i < len; i++) {
-    caml_ba_unsafe_set_1(dst, dstoff + i, string_unsafe_get(src, src_off + i));
+    caml_ba_set_1(dst, dstoff + i, string_unsafe_get(src, src_off + i));
```
Comment (martindemello) [inline lib/runtime.js]: typo: s/dstoff/dst_off

File: lib/runtime.js [hunk 2]
```diff
@@ -31,20 +31,26 @@
...
+//Requires: string_unsafe_set, caml_ba_get_1
 function angstrom_bigstring_blit_to_bytes(src, src_off, dst, dst_off, len) {
   for (var i = 0; i < len; i++) {
-    string_unsafe_set(dst, dst_off + i, caml_ba_unsafe_ref_1(src, src_off + i));
+    string_unsafe_set(dst, dst_off + i, caml_ba_get_1(src, src_off + i));
   }
 }
 
+//Provides: angstrom_bigstring_blit_to_bigstring
+//Requires: caml_ba_unsafe_set_1, caml_ba_get_1
```
Comment (martindemello) [inline lib/runtime.js]: should be caml_ba_set_1 here too

**Comment by martindemello** [comment]
that fixes the `caml_ba_*` issue, but it's still missing the `string_unsafe_*` primitives (i think they got deprecated from ocaml)

**Comment by seliopou** [comment]
I tried your test program and everything seems to now work.

**Comment by martindemello** [comment]
awesome :) thanks for the quick fix

**Pull request merged**